arduino_unit_tests/cc1120_spi.cpp

```diff
@@ -0,0 +1,448 @@
+#include "cc1120_spi.h"
+#include "cc1120_regs.h"
+#include "cc1120_ext_regs.h"
+#include "arduino_unit_tests.h"
+#include "Arduino.h"
+#include <stdint.h>
+#include <SPI.h>
+
+/**
+ * @brief Reads a register from the CC1120.
+ * 
+ * @param addr - The address of the register to read.
+ * @param data - Pointer to the variable to store the read data.
+ * @return true - If the read was successful.
+ * @return false - If the register is not valid, or the status byte is invalid.
+ */
+bool arduinoReadSPI(uint8_t addr, uint8_t *data) {
+    if(addr >= EXT_ADDR) {
+        Serial.println("Not a valid register!");
+        return false;
+    }
+
+    union cc_st ccstatus;
+    digitalWrite(CS, LOW);
+    if (!sendByteReceiveStatus(R_BIT | addr)) {
+        digitalWrite(CS, HIGH);
+        return false;
+    }
+    *data = SPI.transfer(0x00);
+    digitalWrite(CS, HIGH);
+    return true;
+}
+
+/**
+ * @brief - Reads consecutive registers from the CC1120 in burst access mode.
+ * 
+ * @param addr - The address of the first register to read.
+ * @param data - The array to store the read data.
+ * @param len - The number of registers to read.
+ * @return true - If the read was successful.
```

Comment:
Improvement: make a specific return typedef so you can add more status codes in the future